components/CategoryPage.js

```diff
@@ -62,6 +67,54 @@ function CategoryPage({selectedCategory, categories, products}) {
         });
     };
 
+    const totalPages = Math.ceil(products.length / recordsPerPage);
+
+    const renderPaginationControls = () => {
+        return (
+            <nav aria-label="Page navigation example">
+                <ul className="pagination">
+                    {currentPage > 1 && (
+                        <li className="page-item">
+                            <a
+                                className="page-link text-dark"
+                                href="#"
+                                onClick={() => setCurrentPage(currentPage - 1)}
```

Comment:
Hii inakaa poa ukiclose the tag in the last line badili ya ku close in a new line. Code inakuwa easier to read na pia ni more easier ku avoid mistakes
 onClick={() => setCurrentPage(currentPage - 1)}>